File: businesslib/src/main/java/com/wujia/businesslib/event/EventMsg.java
package com.wujia.businesslib.event;

import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

/**
 * author ：shenbingkai@163.com
 * date ：2019-03-31
 * description ：
 */
public class EventMsg extends AbsEventEntiry<EventMsg> {

    public static final int TYPE_NEW_MSG = 1;
    public static final int TYPE_READ = 2;

    public int type = 0;

    public EventMsg() {

    }

    public EventMsg(int type) {
        this.type = type;
    }

    public EventMsg(IMiessageInvoke<EventMsg> invoke) {
        super(invoke);
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onEventBus(EventMsg event) {
        invoke(event);
    }
}

File: businesslib/src/main/java/com/wujia/businesslib/event/EventMsg.kt
package com.wujia.businesslib.event

import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

/**
 * author ：shenbingkai@163.com
 * date ：2019-03-31
 * description ：
 */
class EventMsg : AbsEventEntiry<EventMsg> {

    var type = 0

    constructor(type: Int) {
        this.type = type
    }

    constructor(invoke: IMiessageInvoke<EventMsg>) : super(invoke)

    @Subscribe(threadMode = ThreadMode.MAIN)
    override fun onEventBus(event: EventMsg) {
        invoke(event)
    }

    companion object {

        const val TYPE_NEW_MSG = 1
        const val TYPE_READ = 2
    }
}
